### DISCUSSION TITLE
Avalonia for Visual Studio Code

### DISCUSSION THREAD

**Discussion by prashantvc** [AvaloniaUI/Avalonia#12337] [General]
Upd: Extension is now publicly available, check https://github.com/AvaloniaUI/AvaloniaVSCode

---------------------

Two weeks ago we showed a [sneak peak](https://github.com/AvaloniaUI/Avalonia/discussions/12128) of Avalonia for Visual Studio Code. The extension will significantly enhance the Avalonia development experience in Visual Studio Code. It features our XAML code completion and integrated XAML previewer, making it easier than ever to build beautiful Avalonia apps. 

Today, we are excited to invite you, the sponsors and the members of OpenCollective to try the (extremely) early build of the extension!

Please sign up for the early access [here](https://forms.gle/UUmEdEVpMpET3pvh8), we will start sending out instructions starting tomorrow, 27 July

## What’s in the extension?

The extension aims to bring the most common workflows for avalonia developers. However, the preview will enable the two most common scenarios to kick off the support for VSCode

## Integrated XAML Previewer

The Avalonia XAML Previewer is a powerful tool that allows you to visualise XAML code in real-time. It enables you to efficiently design and fine-tune user interfaces, helping to identify potential issues and visualise the impact of changes instantly.

![telegram-cloud-photo-size-4-5909012943981362401-y](https://github.com/AvaloniaUI/Avalonia/assets/328076/5c2799a4-6ab9-4a09-b4b4-137b521824d0)

## XAML Code completion

The Avalonia XAML in the VS Code is powered by the same code completion engine available for Visual Studio on Windows.

<img width="665" alt="AutoCompleteRM" src="https://github.com/AvaloniaUI/Avalonia/assets/328076/07552979-a055-4db9-95ce-c3ed3ad383ce">

Rich syntax highlighter and contextual code complete will make it lot easier to read and write XAML files

## Feedback
Please share your suggestions, feature requests and report issues for the extension through VS Code’s Help > Report Issue dialog

<img width="700" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/328076/61a1d2cd-ce64-4013-87f2-3bdb6312239e">

You can follow all the suggestions and issues the extension [here](https://github.com/avaloniaUI/Avalonia-vscode-extension/issues).

**Comment by prashantvc** [comment]
Quick update:

Currently investigating an issue (https://github.com/AvaloniaUI/Avalonia-VSCode-Extension/issues/3) where the extension fails to activate on Ubuntu, or crashes other extensions. 

I am working on the fix, please refrain from using the extension on Ubuntu

**Comment by CartBlanche** [comment]
Very cool @prashantvc !! Is the extension all TypeScript?

**Reply by prashantvc** [reply]
It's a mix of TypeScript and C#. 
The core extension is in Typescript, the XAML previewer and Language Server Protocol servers are in C#

VS Code team did a great job with extension APIs, and documentation. It made my work lot easier

**Comment by bsampica** [comment]
Hey guys, I'm looking for access to this.  I'm currently developing in Rider on MacOS, but Visual Studio and VS Code on Windows and Mac are my preferred development environments.  Any chance of getting access to the preview extension?  I filled out the form and sent it in, but I don't think I've heard anything back.

**Reply by prashantvc** [reply]
@MikeCodesDotNET ^

**Reply by timunie** [reply]
> Hey guys, I'm looking for access to this.  I'm currently developing in Rider on MacOS, but Visual Studio and VS Code on Windows and Mac are my preferred development environments.  Any chance of getting access to the preview extension?  I filled out the form and sent it in, but I don't think I've heard anything back.

If you are a sponsor you should have access. We will double check.

**Reply by timunie** [reply]
@bsampica looks like you are not a sponsor, at least we didn't find you. 

In case you're a sponsor: You should have got a verification mail from the team. Please reply to it. Maybe it's in your spam folder by accident, so please also check this. 

Thx.

**Reply by Mrxx99** [reply]
Can newer versions be distributed to the sponsors? I still have 0.0.5

**Reply by timunie** [reply]
Don't know the current version but if there's an update, you should recieve it for sure.

**Reply by bsampica** [reply]
Looks like I never pulled the trigger on being a sponsor, but I'd be happy to, I love this project.  Do all sponsorship levels receive early access, and only monthly, or one time sponsors also?

**Reply by timunie** [reply]
It's the gesture that counts. So whatever you can effort would be ♥️

**Reply by bsampica** [reply]
Done.  :)

**Comment by bsampica** [comment]
I went and checked through my junk email folder, and didn't come up with anything.  If the invitation email could be resent, I'll keep an eye out for it.

**Reply by timunie** [reply]
It was Sunday yesterday and you weren't a sponsor until yesterday. I think it's fair enough to give us some time to check things.

**Reply by bsampica** [reply]
Yea man, no big deal.  When you get a chance.

**Reply by timunie** [reply]
should be done if I am informed correctly.

**Comment by bsampica** [comment]
I'm not sure what I'm looking for.   I haven't received an email invitation that I can find.  Can you tell me who it'd be coming from?
________________________________

**Reply by MikeCodesDotNET** [reply]
It was sent 7 hours from team@avaloniaui.net.

**Comment by bsampica** [comment]
Thanks Mike, I don't see anything.  I've checked junk, deleted, the works.  Can you please resend it if it's not too much trouble?
________________________________

**Comment by bsampica** [comment]
Nevermind, sorted...was in my gmail.
________________________________

**Comment by PapaSmurf-ML64** [comment]
Also awaiting access to the preview, if you could take a look.

**Reply by MikeCodesDotNET** [reply]
It’s now publicly available. You can find it if you search for it in the marketplace.

**Comment by NCLnclNCL** [comment]
good

**Comment by waliarubal** [comment]
I am wondering if this extension is available now for general public? I am willing to start an open source project for Ubuntu using Avalonia UI. Is it available so that I can do development on Ubuntu machine?

**Reply by waliarubal** [reply]
Nevermind, read last comment. I will check marketplace.
